src/RestApi/StoreApi/Schemas/CartItemSchema.php

```diff
@@ -242,6 +248,31 @@ public function get_item_response( $cart_item ) {
 		];
 	}
 
+	/**
+	 * Generate a summary from the product short/full description.
+	 *
+	 * @param \WC_Product $product Product instance.
+	 * @return string
+	 */
+	protected function get_product_summary( \WC_Product $product ) {
+		$short_description = $product->get_short_description();
+
+		if ( $short_description ) {
+			return \wc_format_content( $short_description );
+		}
+
+		$description = $product->get_description();
+		$length      = \strlen( \wp_strip_all_tags( $description ) );
+
+		// Try to extract the first sentence only if the description is too long.
+		if ( 150 < $length ) {
+			$description_p = \wpautop( $description );
+			$description   = \strstr( $description_p, '</p>' ) ? \substr( $description_p, 0, \strpos( $description_p, '</p>' ) + 4 ) : \wc_trim_string( $description, 150 );
+		}
```

Comment:
Would be good to extract this to a helper and test it on various content so you can verify it works as expected with what gets thrown at it.

Reply:
I've moved this into a utility class and added some tests to ensure things get trimmed.

I also found a nice WP function to trim words rather than letters which seems to create nicer summaries. implemented that too.